prime-dotnet-webapi/Services/PartyService.cs

```diff
@@ -201,5 +206,40 @@ private IQueryable<Party> GetBasePartyQuery()
                 .Include(p => p.PartyEnrolments)
                 .Include(p => p.PartyCertifications);
         }
+
+        public async Task<int> UpdatePartyHpdid(int limit)
+        {
+            //query party ID where HPDID is null
+            var partyIds = await _context.Parties
+                .Where(p => p.HPDID == null)
+                .OrderBy(p => p.Id)
+                .Select(p => p.Id)
+                .Take(limit)
+                .ToListAsync();
+
+            var counter = 0;
+            foreach (var pId in partyIds)
+            {
+                //for each party ID, get the party record
+                var party = await _context.Parties.Where(p => p.Id == pId).FirstOrDefaultAsync();
+                //get the user object from keycloak
+                var party_user = await _keycloakClient.GetUser(party.UserId);
```

Comment:
This needs to handle the case where cannot find user in Keycloak ... log message and skip to next?

Also variable names should use Camel case, as per rest of prime-dotnet-webapi code base

Reply:
Good point. Updated the code accordingly.